gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionTransformer.scala

```diff
@@ -16,17 +16,74 @@
  */
 package org.apache.gluten.expression
 
-import org.apache.gluten.substrait.expression.ExpressionNode
+import org.apache.gluten.substrait.expression.{ExpressionBuilder, ExpressionNode}
 
-import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.expressions.{Expression, Literal}
 import org.apache.spark.sql.types.DataType
 
-trait ExpressionTransformer {
-  def doTransform(args: java.lang.Object): ExpressionNode
-  def dataType: DataType
-}
+import scala.collection.JavaConverters._
+
+// ==== Expression transformer basic interface start ====
 
-trait ExpressionTransformerWithOrigin extends ExpressionTransformer {
+trait ExpressionTransformer {
+  def substraitExprName: String
+  def children: Seq[ExpressionTransformer]
   def original: Expression
   def dataType: DataType = original.dataType
+  def nullable: Boolean = original.nullable
+
+  def doTransform(args: java.lang.Object): ExpressionNode = {
+    val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
+    // TODO: the funcName seems can be simplified to `substraitExprName`
+    val funcName: String =
+      ConverterUtils.makeFuncName(substraitExprName, original.children.map(_.dataType))
```

Comment:
Just highlight the implicit change, the function config of function name is never been used. The default function config of makeFuncName is `NON`.